packages/@aws-cdk/tmp-toolkit-helpers/src/api/refactoring/digest.ts

```diff
@@ -82,10 +89,29 @@ export function computeResourceDigests(template: CloudFormationTemplate): Record
   const result: Record<string, string> = {};
   for (const id of order) {
     const resource = resources[id];
-    const depDigests = Array.from(graph[id]).map((d) => result[d]);
-    const propsWithoutRefs = hashObject(stripReferences(stripConstructPath(resource)));
-    const toHash = resource.Type + propsWithoutRefs + depDigests.join('');
-    result[id] = crypto.createHash('sha256').update(toHash).digest('hex');
+
+    const model = loadResourceModel(resource.Type);
+    const primaryIdentifier = intersection(Object.keys(resource.Properties ?? {}), model?.primaryIdentifier ?? []);
+    if (primaryIdentifier.length > 0) {
+      // The resource has a physical ID defined, so we can
+      // use the ID and the type as the identity of the resource.
+
+      const toHash =
+        resource.Type +
+        primaryIdentifier
+          .sort()
+          .map((attr) => JSON.stringify(resource.Properties[attr]))
+          .join('');
+      result[id] = crypto.createHash('sha256').update(toHash).digest('hex');
```

Comment:
💅 The exact hash calculation is the same in both cases so could be factored out of the `if`.